monai/transforms/utility/dictionary.py

```diff
@@ -1419,6 +1423,53 @@ def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> dict[Hashable, N
         return d
 
 
+class TorchIOd(MapTransform, RandomizableTrait):
+    """
+    Dictionary-based wrapper of :py:class:`monai.transforms.TorchIO` for transforms.
+    All transforms in TorchIO can be applied randomly with probability p by specifying the `p=` argument.
+    """
+
+    backend = TorchIO.backend
+
+    def __init__(
+        self,
+        keys: KeysCollection,
+        name: str,
+        apply_same_transform: bool = False,
```

Comment:
The same comments about the RandomizableTrait inherit with the array version.
And what's this `apply_same_transform` arg used for? In MONAI, we always apply the same transform on the  sequence keys, such as image and label. Is there a typical case here users may want apply different transforms on the different keys? If not, perhaps we can remove this one.

Reply:
Apply same is indeed a bit misleading.

I had a use case with random affine, where the same random affine should be applied to both input and output for augmentation with patches.

Same/similar for random elastic deformations. 

Can remove if you want to, would constrain the applicable augmentations in a sense or restrict the net to learn to output only a certain orientation/affine, i.e. all inputs would need registration to the data with which the net was trained.

Reply:
It's different in TorchIO so I had to hack arround it. The `apply_same = false` is merely to reproduce TorchIOs behavior.